src/libraries/System.Reflection.Metadata/src/System/Reflection/Internal/MemoryBlocks/ExternalMemoryBlockProvider.cs

```diff
@@ -36,7 +36,7 @@ protected override AbstractMemoryBlock GetMemoryBlockImpl(int start, int size)
         public override Stream GetStream(out StreamConstraints constraints)
         {
             constraints = new StreamConstraints(null, 0, _size);
-            return new ReadOnlyUnmanagedMemoryStream(_memory, _size);
+            return new UnmanagedMemoryStream(_memory, _size);
```

Comment:
The overload with two parameters [opens](https://learn.microsoft.com/en-us/dotnet/api/system.io.unmanagedmemorystream.-ctor?view=net-9.0#system-io-unmanagedmemorystream-ctor(system-byte*-system-int64)) the stream in read-only mode.